### PR TITLE
Treasure map filters fix

### PR DESCRIPTION
**Pull request by Theodlz**
This PR does 2 things:
- Ensure we post correct filters to treasuremap's endpoint.
- makes sure in multiple methods in both backend and frontend, that filters submitted are in the instrument's sncosmo filters list.

I haven't tested the actual submission to treasuremap (i need to check if they have a sandbox), so a second PR might be needed. Or just giving a little more time to finish this one.

### CODE REVIEW

File: static/js/components/NewInstrument.jsx [hunk 1]
```diff
@@ -226,6 +226,11 @@ const NewInstrument = () => {
         title:
           "Configuration data i.e. {'overhead_per_exposure': 2.0, 'readout': 8.0, 'slew_rate': 2.6, 'filt_change_time': 60.0}",
       },
+      treasuremap_id: {
+        type: "integer",
+        title: "Treasuremap ID",
+        description: "ID of the instrument to submit to Treasuremap (optional)",
```
Comment (mcoughlin) [inline static/js/components/NewInstrument.jsx:232]: I guess we already have ZTF and DECam and a few others hardcoded somewhere. Can you put them by your FILTERS list for easy finding?

**Review by mcoughlin — APPROVED** [review]
This is looking good. Might need to fix the gwemopt tests with these new filters.

File: static/js/components/NewInstrument.jsx (hunk 1 above)
Comment (Theodlz) [reply static/js/components/NewInstrument.jsx:232]: You mean just to keep track of it? Or to try use that list when not added to the instrument in the DB?

Comment (mcoughlin) [reply static/js/components/NewInstrument.jsx:232]: Yeah the latter is what I am thinking. I am not sure if we have done a good job mapping things, but I might be wrong.

Comment (Theodlz) [reply static/js/components/NewInstrument.jsx:232]: We havent mapped the ids anywhere. They are in the instruments model, and one can post it to the DB. We just have it added to ZTF in the demo data. But I can add a mapper for sure.

Comment (mcoughlin) [reply static/js/components/NewInstrument.jsx:232]: Ok that might be good then. We want to be able to add it to the instruments model of course, but nothing wrong with a few common ones.

Comment (Theodlz) [reply static/js/components/NewInstrument.jsx:232]: I added the mapper with a bunch of instruments we have on Fritz, and some common ones that might be on other instances.

Comment (mcoughlin) [reply static/js/components/NewInstrument.jsx:232]: Seems good. should be good to go once tested.

**Comment by Theodlz** [comment]
@mcoughlin with this, I've successfully submitted pointings (planned and completed, meaning obs plans and executed observations) to treasuremap. 

Also, I used this as an opportunity to add code to look in the right partitions directly in the sources, observations, and candidates handlers.

**Comment by Theodlz** [comment]
@mcoughlin you might want to double check what as used as bandwidth for the filters that don't have a filter name in treasuremap (uvot::white), otherwise should be good to go in once tested.

**Comment by mcoughlin** [comment]
@Theodlz What about getting it from the bandpass itself:

>>> from sncosmo.bandpasses import _BANDPASSES
>>> existing_bandpasses_names = [val['name'] for val in _BANDPASSES.get_loaders_metadata()]
>>> existing_bandpasses_names
['bessellux', 'bessellb', 'bessellv', 'bessellr', 'besselli', 'standard::u', 'standard::b', 'standard::v', 'standard::r', 'standard::i', 'desu', 'desg', 'desr', 'desi', 'desz', 'desy', 'sdssu', 'sdssg', 'sdssr', 'sdssi', 'sdssz', 'f435w', 'f475w', 'f555w', 'f606w', 'f625w', 'f775w', 'f850lp', 'nicf110w', 'nicf160w', 'f098m', 'f105w', 'f110w', 'f125w', 'f127m', 'f139m', 'f140w', 'f153m', 'f160w', 'f218w', 'f225w', 'f275w', 'f300x', 'f336w', 'f350lp', 'f390w', 'f689m', 'f763m', 'f845m', 'f438w', 'uvf475w', 'uvf555w', 'uvf606w', 'uvf625w', 'uvf775w', 'uvf814w', 'uvf850lp', 'kepler', 'cspb', 'csphs', 'csphd', 'cspjs', 'cspjd', 'cspv3009', 'cspv3014', 'cspv9844', 'cspys', 'cspyd', 'cspg', 'cspi', 'cspk', 'cspr', 'cspu', 'f070w', 'f090w', 'f115w', 'f150w', 'f200w', 'f277w', 'f356w', 'f444w', 'f140m', 'f162m', 'f182m', 'f210m', 'f250m', 'f300m', 'f335m', 'f360m', 'f410m', 'f430m', 'f460m', 'f480m', 'f560w', 'f770w', 'f1000w', 'f1130w', 'f1280w', 'f1500w', 'f1800w', 'f2100w', 'f2550w', 'f1065c', 'f1140c', 'f1550c', 'f2300c', 'lsstu', 'lsstg', 'lsstr', 'lssti', 'lsstz', 'lssty', 'keplercam::us', 'keplercam::b', 'keplercam::v', 'keplercam::r', 'keplercam::i', '4shooter2::us', '4shooter2::b', '4shooter2::v', '4shooter2::r', '4shooter2::i', 'f062', 'f087', 'f106', 'f129', 'f158', 'f184', 'f213', 'f146', 'ztfg', 'ztfr', 'ztfi', 'uvot::b', 'uvot::u', 'uvot::uvm2', 'uvot::uvw1', 'uvot::uvw2', 'uvot::v', 'uvot::white', 'ps1::open', 'ps1::g', 'ps1::r', 'ps1::i', 'ps1::z', 'ps1::y', 'ps1::w', 'atlasc', 'atlaso', '2massj', '2massh', '2massks', 'gaia::gbp', 'gaia::g', 'gaia::grp', 'gaia::grvs', 'tess']
>>> band = sncosmo.get_bandpass('bessellux')
>>> band
<Bandpass 'bessellux' at 0x10d9a22b0>
>>> (band.minwave()+band.maxwave())/2
3600.0
>>> (band.maxwave()-band.maxwave())
0.0
>>> (band.maxwave()-band.minwave())
1200.0
>>> (band.maxwave()-band.minwave())
1200.0
>>> 

And can just set it like this for any missing bandpasses?

**Comment by Theodlz** [comment]
done @mcoughlin. Let me just wait for the next test event to come in so I can verify that it still works (had to flush my db when changes branches).

**Comment by mcoughlin** [comment]
@Theodlz Ok feel free to merge when comfortable.

**Comment by Theodlz** [comment]
failures seem unrelated. I can still generate summaries and create observation plans with galaxy targeting. Maybe a caching issue? Merging

**Pull request merged**